main.js

```diff
@@ -446,6 +317,136 @@ onReady = () => {
     }
 }; /* onReady() */
 
+// initialise the db
```

Comment:
Your PRs are really hard to review, as you move huge code blocks around, which make it impossible for me to see what YOU changed